src/hooks/useGeolocationHandler.ts

```diff
@@ -0,0 +1,82 @@
+
+import { Matcher, SelectableStaticFilter, useSearchActions, useSearchState } from '@yext/search-headless-react';
+import { executeSearch } from '../utils/search-operations';
+import { getUserLocation } from '../utils/location-operations';
+import { useCallback } from 'react';
+
+const LOCATION_FIELD_ID = 'builtin.location';
+const METERS_PER_MILE = 1609.344;
+
+/**
+ * The props for {@link useGeolocationHandler} hook.
+ *
+ * @internal
+ */
+interface GeolocationhandlerProps {
+  /** A React dispatch function use to update the status of fetching user's location. */
+  setIsFetchingLocation: React.Dispatch<React.SetStateAction<boolean>>,
+  /** Configuration used when collecting the user's location. */
+  geolocationOptions?: PositionOptions,
+  /**
+   * The radius, in miles, around the user's location to find results. Defaults to 50.
+   * If location accuracy is low, a larger radius may be used automatically.
+   */
+  radius?: number,
+  /** Custom handler function to call after user's position is successfully determined. */
+  handleUserPosition?: (position: GeolocationPosition) => void
```

Comment:
Small nit: I'm not wild about the names `handleUserPosition` or `setIsFetchingLocation`. I think the latter is a bit long and convoluted. The former is a bit vague: `handleUserPosition` doesn't sound like a callback that fires when the user's position is obtained.

Reply:
`setIsFetchingLocation` is following the common pattern -- `setIsBlah` -- we use for a **boolean setter** function of react `useState` (e.g. setIsFeedbackProvided or setIsOptionsDisabled) so I thought the naming here would make things consistent. Outside of the pattern, maybe `setIsUserLocationFound` or `updateGetLocationStatus` but I don't think it conveys that it can be set to a boolean true, regardless if the location was determined successfully or not, as long as the fetching process is done? Similarly with our callbacks `handleBlah` pattern. Let me know if you have other names you prefer!

Reply:
I don't totally agree with the idea that we should never use the word "on" and always use "handle". I think if done correctly, we could use both and each could convey a different meaning. But, that's a higher-level team convo.